src/rulesets/asyncapi/index.json

```diff
@@ -0,0 +1,398 @@
+{
+  "formats": [
+    "asyncapi2"
+  ],
+  "rules": {
+    "asyncapi2-channel-no-empty-parameter": {
+      "description": "Channel path should not have empty parameter substitution pattern.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": "{}"
+        }
+      }
+    },
+    "asyncapi2-channel-no-query-nor-fragment": {
+      "description": "Channel path should not include a query (`?`) or a fragment (`#`) delimiter.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": "[\\?#]"
+        }
+      }
+    },
+    "asyncapi2-channel-no-trailing-slash": {
+      "description": "Channel path should not end with a slash.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.channels.*~",
+      "then": {
+        "function": "pattern",
+        "functionOptions": {
+          "notMatch": ".+\\/$"
+        }
+      }
+    },
+    "asyncapi2-headers-schema-type-object": {
+      "description": "Headers schema type should be `object`.",
+      "message": "Headers schema type should be `object` ({{error}}).",
+      "severity": 0,
+      "recommended": true,
+      "type": "validation",
+      "given": [
+        "$.components.messageTraits.*.headers",
+        "$.components.messages.*.headers",
+        "$.channels.*.[publish,subscribe].message.headers"
+      ],
+      "then": {
+        "function": "schema",
+        "functionOptions": {
+          "schema": {
+            "type": "object",
+            "properties": {
+              "type": {
+                "enum": [
+                  "object"
+                ]
+              }
+            },
+            "required": [
+              "type"
+            ]
+          }
+        },
+        "allErrors": true
+      }
+    },
+    "asyncapi2-info-contact-properties": {
+      "description": "Contact object should have `name`, `url` and `email`.",
+      "recommended": true,
+      "type": "style",
+      "given": "$.info.contact",
+      "then": [
+        {
+          "field": "name",
+          "function": "truthy"
+        },
+        {
+          "field": "url",
+          "function": "truthy"
+        },
+        {
+          "field": "email",
+          "function": "truthy"
+        }
+      ]
+    },
+    "asyncapi2-info-contact": {
+      "description": "Info object should contain `contact` object.",
+      "recommended": true,
+      "type": "style",
```

Comment:
@P0lip I've been wondering for a while if we are misusing "type": "style". I think a style recommendation is "use camelCase" and other stylistic only things. Wanting contact information seems more functional than just style. 

Did we ever have a particular definition for when something is style or not, and do we want to figure it out before we bake it into async too?